src/app/icd/client/DefaultICDClientStorage.cpp

```diff
@@ -471,7 +489,23 @@ CHIP_ERROR DefaultICDClientStorage::DeleteAllEntries(FabricIndex fabricIndex)
     }
     ReturnErrorOnFailure(
         mpClientInfoStore->SyncDeleteKeyValue(DefaultStorageKeyAllocator::ICDClientInfoKey(fabricIndex).KeyName()));
-    return mpClientInfoStore->SyncDeleteKeyValue(DefaultStorageKeyAllocator::FabricICDClientInfoCounter(fabricIndex).KeyName());
+    ReturnErrorOnFailure(
+        mpClientInfoStore->SyncDeleteKeyValue(DefaultStorageKeyAllocator::FabricICDClientInfoCounter(fabricIndex).KeyName()));
+
+    for (auto fabric = mFabricList.begin(); fabric != mFabricList.end(); fabric++)
+    {
+        if (*fabric == fabricIndex)
+        {
+            fabric = mFabricList.erase(fabric);
```

Comment:
Why do we need to assign to `fabric` here?

Reply:
removed, thanks